contracts/workflows/RoyaltyTokenDistributionWorkflows.sol

```diff
@@ -230,7 +229,8 @@ contract RoyaltyTokenDistributionWorkflows is
             licenseTermsData: licenseTermsData
         });
 
-        ipRoyaltyVault = _deployRoyaltyVault(ipId, msg.sender);
+        ipRoyaltyVault = ROYALTY_MODULE.ipRoyaltyVaults(ipId);
+        ipRoyaltyVault = (ipRoyaltyVault == address(0)) ? ROYALTY_MODULE.deployVault(ipId) : ipRoyaltyVault;
```

Comment:
do we need to rename this function `registerIpAndAttachPILTermsAndDeployRoyaltyVault` to `registerIpAndAttachPILTerms`?

Reply:
per discussion, we still might deploy vault in the function, hence keep the same name.